### PR TITLE
Search by title

### PR DESCRIPTION
**Pull request by felipesere**
This PR allows the search API endpoint to find books by the title.

Consider:
* the code has not yet been hooked into the UI
* modules around the `SearchByIsbn` where renamed to Finder and friends

Todo:
* [x] add the necessary JS to use the search

### CODE REVIEW

File: lib/faros/books/google_finder.ex [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    |> HTTPotion.get
+    |> Map.get(:body)
+    |> parse
+  end
+
+  def sanitise_title(title) do
+    URI.encode_www_form(title)
+  end
+
+  defp sanitise(isbn) do
```
Comment (marosluuce) [inline lib/faros/books/google_finder.ex]: This should probably be `sanitise_isbn`. Maybe we need to extract some common ways to sanitise?

File: lib/faros/books/google_finder.ex [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    String.replace(isbn, "-", "")
+  end
+
+  def build_url(thing, kind) do
+    "https://www.googleapis.com/books/v1/volumes\?q\=#{query(kind)}:#{thing}"
```
Comment (marosluuce) [inline lib/faros/books/google_finder.ex]: Is it worth putting this url into a config?

File: spec/isbn-lookup_spec.js [hunk 3]
```diff
@@ -27,6 +48,13 @@ describe('Isbn-Lookup', () => {
     expect($('[data-id="book-description"]').val()).toEqual('Bla Bla Bla')
     expect($('[data-id="book-link"]').val()).toEqual('http://example.com')
   })
+
+  function captureUrlOn(isbnLookup) {
```
Comment (marosluuce) [inline spec/isbn-lookup_spec.js]: You could do something like:
```
let fake = (arg, callback) => arg
spyOn($, 'get').and.callFake(fake)
lookup.link.click()
return fake()
```

File: web/books/controller.ex [hunk 4]
```diff
@@ -53,8 +53,19 @@ defmodule Faros.Books.Controller do
     redirect conn, to: "/books"
   end
 
-  def lookup(conn, %{"isbn" => isbn}) do
-    case SearchByIsbn.execute(isbn) do
+  def lookup(conn, query) do
+    isbn = query["isbn"]
```
Comment (marosluuce) [inline web/books/controller.ex]: What if you used `fetch("isbn")`?

Then you could say:
```
case {isbn, title} do
  {:error, :error} -> ...
  {:error, {:ok, title}} -> ...
  {{:ok, isbn}, _ } -> ...
end
```

Not sure if that looks better than empty strings. Or are you guaranteed to get empty strings. If so, you could move the destructuring into the signature.

File: lib/faros/books/google_finder.ex (hunk 2 above)
Comment (felipesere) [inline lib/faros/books/google_finder.ex]: The URL is very specific to the GoogleFinder and won't be replaced in any kind of "staging" environment... Not sure if it is worth yet.

**Comment by marosluuce** [comment]
:ok:

**Pull request merged**